tests/ztest/include/ztest_test.h

```diff
@@ -139,8 +141,17 @@ static inline void unit_test_noop(void)
  *
  * @param name Name of the testing suite
  */
+
+/* definitions for use with testing application shared memory   */
+#ifdef APP_SHARED_MEM
```

Comment:
Maybe CONFIG_APP_SHARED_MEM can be used instead of this. I feel its easier to maintain.